src/PointwiseFunctions/AnalyticData/CurvedWaveEquation/ScalarWaveGr.cpp

```diff
@@ -0,0 +1,145 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "PointwiseFunctions/AnalyticData/CurvedWaveEquation/ScalarWaveGr.hpp"
+
+#include <pup.h>
+
+#include "DataStructures/DataBox/Prefixes.hpp"
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/EagerMath/DotProduct.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "DataStructures/Variables.hpp"
+#include "Domain/CoordinateMaps/Affine.hpp"
+#include "Domain/CoordinateMaps/CoordinateMap.hpp"
+#include "Domain/CoordinateMaps/ProductMaps.hpp"
+#include "Domain/Element.hpp"
+#include "Domain/LogicalCoordinates.hpp"
+#include "Domain/Mesh.hpp"
+#include "ErrorHandling/Assert.hpp"
+#include "Evolution/Systems/CurvedScalarWave/Equations.hpp"
+#include "NumericalAlgorithms/Spectral/Spectral.hpp"
+#include "Parallel/PupStlCpp11.hpp"
+#include "PointwiseFunctions/GeneralRelativity/Christoffel.hpp"
+#include "PointwiseFunctions/GeneralRelativity/ComputeSpacetimeQuantities.hpp"
+#include "PointwiseFunctions/GeneralRelativity/IndexManipulation.hpp"
+#include "PointwiseFunctions/GeneralRelativity/Tags.hpp"
+#include "Utilities/ConstantExpressions.hpp"
+#include "Utilities/ContainerHelpers.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/MakeWithValue.hpp"
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TaggedTuple.hpp"
+#include "Utilities/TypeTraits.hpp"
+
+/// \cond
+namespace CurvedScalarWave ::AnalyticData {
+
+template <typename ScalarFieldData, typename BackgroundGrData>
+void ScalarWaveGr<ScalarFieldData, BackgroundGrData>::pup(PUP::er& p) noexcept {
+  p | flat_space_scalar_wave_data_;
+  p | background_gr_data_;
+}
+
+template <typename ScalarFieldData, typename BackgroundGrData>
+tuples::TaggedTuple<Pi>
+ScalarWaveGr<ScalarFieldData, BackgroundGrData>::variables(
+    const tnsr::I<DataVector, volume_dim>& x, tmpl::list<Pi> /*meta*/) const
+    noexcept {
+  constexpr double default_initial_time = 0.;
+  const auto flat_space_scalar_wave_vars =
+      flat_space_scalar_wave_data_.variables(
+          x, default_initial_time,
+          tmpl::list<ScalarWave::Pi, ScalarWave::Phi<volume_dim>,
+                     ScalarWave::Psi>{});
+  const auto flat_space_dt_scalar_wave_vars =
```

Comment:
don't use the time derivatives; just use Pi from the flat-space solution/data

Reply:
changed